File: src/main/java/org/yggdrasil/core/ledger/transaction/TransactionInput.java
package org.yggdrasil.core.ledger.transaction;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.io.Serializable;
import java.math.BigDecimal;

@JsonInclude
public class TransactionInput implements Serializable {

    // Reference to a transaction's output
    protected final TransactionOutPoint txnOutPt;
    protected final BigDecimal value;


    public TransactionInput(){
        this.txnOutPt = null;
        this.value = BigDecimal.valueOf(-1);
    }

    public TransactionInput(TransactionOutPoint txnOut, BigDecimal value){
        this.txnOutPt = txnOut;
        this.value = value;
    }

    public TransactionInput(byte[] prevBlkHash, byte[] prevTxHash,  BigDecimal valueOut){
        this.txnOutPt = new TransactionOutPoint(prevBlkHash, prevTxHash, valueOut);
        this.value = valueOut;
    }

    public TransactionOutPoint getTxnOutPt() {
        return txnOutPt;
    }

    public BigDecimal getValue() {
        return value;
    }
}

File: yggdrasil-core-csharp/yggdrasil-core/core/ledger/transaction/TransactionInput.cs
namespace yggdrasil_core.core.ledger.transaction
{
    public class TransactionInput : LedgerHashableItem
    {
        private readonly TransactionOutPoint _txOutPt;
        public TransactionOutPoint TxOutPt { get { return _txOutPt; } }

        private readonly Coin _value;
        public Coin Value { get { return _value; } }

        public TransactionInput(TransactionOutPoint txOutPt, Coin value)
        {
            this._txOutPt = txOutPt;
            this._value = value;
        }

        public TransactionInput(byte[] prevBlkHash, byte[] prevTxHash, Coin valueOut)
        {
            this._txOutPt = new TransactionOutPoint(prevTxHash, prevTxHash, valueOut);
            this._value = valueOut;
        }

        public byte[] Bytes()
        {
            byte[] bytes = new byte[0];
            bytes = (byte[])bytes.Concat(_txOutPt.Bytes());
            bytes = (byte[])bytes.Concat(_value.Bytes());
            return bytes;
        }
    }
}
